src/util/favicon.js

```diff
@@ -0,0 +1,89 @@
+import { DOMParser } from "@xmldom/xmldom";
+import isUrl from "is-url";
+
+/**
+ * Parse the HTML source into a DOM document
+ *
+ * @param {string | undefined} htmlSource the HTML source
+ * @returns {Document | undefined} the DOM document
+ */
+const parseHtmlSource = (htmlSource) => {
+  if (htmlSource && htmlSource.length > 0) {
+    // use a return statement for the error handler to avoid the console warning
+    // as any error will result in fallback favicon
+    return new DOMParser().parseFromString(htmlSource, "text/html");
+  }
+};
+
+/**
+ * Return the favicon URL from the HTML links collection
+ *
+ * @param {HTMLCollectionOf<Element> | undefined} links the HTML links collection
+ * @param {string} origin the origin URL to be used to build the favicon url
+ * @returns {string} the first found favicon URL or empty object if none found
+ */
+const getFaviconUrlFromLinks = (links, origin) => {
+  let faviconURL = "";
+
+  if (links && links.length > 0 && origin) {
+    for (let i in Array.from(links)) {
+      const link = links[i];
+      const rel = link.getAttribute("rel");
+      if (rel && rel.split(" ").includes("icon")) {
+        const href = link.getAttribute("href");
+        if (href) {
+          faviconURL = new URL(href, origin).toString();
```

Comment:
we could remove the toString so that we only deal with URLs until the end. This change has been made in the mobile PR.